op-node/rollup/derive/calldata_source.go

```diff
@@ -61,6 +62,17 @@ type DataSource struct {
 // If there is an error, it will attempt to fetch the result on the next call to `Next`.
 func NewDataSource(ctx context.Context, log log.Logger, cfg *rollup.Config, fetcher L1TransactionFetcher, block eth.BlockID, batcherAddr common.Address) DataIter {
 	_, txs, err := fetcher.InfoAndTxsByHash(ctx, block.Hash)
+	if err != nil {
```

Comment:
Why do we return right away if there is some error?

Reply:
Based on the last line of the function doc, I thought this is the right thing to do here.

```
// If there is an error, it will attempt to fetch the result on the next call to `Next`.
```